agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AppIdSupplier.java

```diff
@@ -75,7 +77,9 @@ public void startAppIdRetrieval() {
     try {
       newTask = new GetAppIdTask(getAppIdUrl(connectionString));
     } catch (MalformedURLException e) {
-      logger.warn(e.getMessage(), e);
+      try (MdcScope ignored = Mdc.APP_ID_ERROR.makeActive()) {
+        logger.warn(e.getMessage(), e);
+      }
```

Comment:
@heyams I pushed a commit to show a nice option of turning the constants into enums combined with using try-with-resource